src/fidesops/api/v1/endpoints/policy_endpoints.py

```diff
@@ -62,6 +63,19 @@ def get_policy_list(
     return paginate(policies, params=params)
 
 
+def get_policy_or_error(db: Session, policy_key: FidesOpsKey) -> Policy:
+    """Helper method to load Policy or throw a 404"""
+    logger.info(f"Finding policy with key '{policy_key}'")
+    policy = Policy.get_by(db=db, field="key", value=policy_key)
+    if not policy:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
```

Comment:
There's a specific `NotFoundException` in `common_exceptions.py.

Reply:
I see - no one's using this though, and I'm not sure I see the benefit of creating a `NotFoundException` - it seems like it obscures the fact that it's an HTTPException.

Reply:
It _does_ inherit from HTTPException, and references the status code. THe other HTTPexceptions (e.g. AuthorizationError, AuthenticationFailure) here seem to be used, I think this is just an inconsistency in our useage - see, e.g. https://github.com/ethyca/fidesops/blob/main/src/fidesops/api/v1/endpoints/oauth_endpoints.py , where we're using the custom `AuthenticationFailure` in the first method, and then a bare `HTTPException` @ line 96. 

). My preference would be to use this everywhere we're throwing a 404, but at the very least I don't think we should delete it.

Reply:
yes! i understand it inherits from HTTPException, but to me it's more useful to see the HTTPException for yourself, rather than look up the custom `NotFoundException` and verify that it will throw an HTTPException.  It's that the `NotFoundException` doesn't add anything new.  

Agreed on inconsistency - in my opinion there's no need to throw the custom `AuthenticationFailure`s either.